app/controllers/generic_phrases_controller.rb

```diff
@@ -2,96 +2,56 @@ class GenericPhrasesController < ApplicationController
   include Searchable
 
   def index
-    options = {
-      page: page,
-      sort_key: sort_key,
-      sort_direction: sort_dir,
-      verb: verb,
-      adjective: adjective,
-    }
-
-    @presenter = GenericPhrasesPresenter.new(generic_phrase_results, verb_results, adjective_results, search_params, options)
+    @presenter = GenericPhrasesPresenter.new(generic_phrase_results, verb_results, adjective_results, search_options)
   end
 
   def show
-    generic_phrase = GenericPhrase.find(params[:id])
+    @generic_phrase = GenericPhrase.find(params[:id])
 
-    @presenter = GenericPhrasePresenter.new(generic_phrase, most_frequent_exact_matches(generic_phrase),
-                                            mentions(generic_phrase), survey_answers(generic_phrase),
-                                            most_frequent_co_occurring_generic_phrases(generic_phrase))
+    @presenter = GenericPhrasePresenter.new(@generic_phrase, most_frequent_exact_matches,
+                                            mentions, survey_answers,
+                                            most_frequent_co_occurring_generic_phrases)
   end
 
 private
 
   def generic_phrase_results
-    options = {
-      sort_key: sort_key,
-      sort_dir: sort_dir,
-    }
-
-    options[:verb] = verb if verb.present?
-    options[:adjective] = adjective if adjective.present?
-
-    GenericPhrase.search(from_date_as_datetime, to_date_as_datetime, options)
-  end
-
-  def verb_results
-    Verb.unique_sorted.map(&:name)
+    @generic_phrase_results ||= GenericPhrase.search(from_date_as_datetime, to_date_as_datetime, search_options)
   end
 
-  def adjective_results
-    Adjective.unique_sorted.map(&:name)
-  end
-
-  def most_frequent_exact_matches(generic_phrase)
-    Phrase
-      .most_frequent_for_generic_phrase(generic_phrase, from_date_as_datetime, to_date_as_datetime)
-      .take(10)
+  def most_frequent_exact_matches
+    @most_frequent_exact_matches ||= begin
+      Phrase
+        .most_frequent_for_generic_phrase(@generic_phrase, from_date_as_datetime, to_date_as_datetime)
+        .take(10)
+    end
   end
 
-  def most_frequent_co_occurring_generic_phrases(generic_phrase)
-    GenericPhrase
-      .most_frequent_co_occurring(generic_phrase, from_date_as_datetime, to_date_as_datetime)
+  def most_frequent_co_occurring_generic_phrases
+    @most_frequent_co_occurring_generic_phrases ||= begin
+      GenericPhrase
+        .most_frequent_co_occurring(@generic_phrase, from_date_as_datetime, to_date_as_datetime)
       .take(10)
```

Comment:
Not sure what the benefit of making these conditional assignments is, as they're only ever called once per view and will be recomputed on every new request for the page. I'm guessing this is more about being consistent.

Reply:
Yeah, more about consistency. In general, expensive API/DB calls are memoized and I'd expect them to be. It doesn't cost anything to memoize and if someone adds code that uses it somewhere they might not realise how expensive it is and increase the load time of pages (I have actually seen this so it's not an abstract fear)

Reply:
Fair point. Some of the memoized calls I've seen area being used but not all, but I'm happy to roll with this for now.